homeassistant/components/http/auth.py

```diff
@@ -235,4 +282,36 @@ async def auth_middleware(
         request[KEY_AUTHENTICATED] = authenticated
         return await handler(request)
 
+    app.middlewares.append(session_middleware(HomeAssistantCookieStorage(hass)))
     app.middlewares.append(auth_middleware)
+
+
+@callback
+def _async_perform_action_on_non_local(
+    request: Request,
+    strict_connection_static_file_content: str | None,
+) -> StreamResponse | None:
+    """Perform strict connection mode action if the request is not local."""
+    try:
+        ip_address_ = ip_address(request.remote)  # type: ignore[arg-type]
+    except ValueError:
+        _LOGGER.debug("Invalid IP address: %s", request.remote)
+        ip_address_ = None
+
+    if ip_address_ and is_local(ip_address_):
+        return None
+
+    _LOGGER.debug("Perform strict connection action for %s", ip_address_)
+    if strict_connection_static_file_content:
```

Comment:
Should we show the static page also for invalid ip addresses?

Reply:
We currently show the static page on a invalid ip address.

See line 299 and 301. We check if the ip address is not None

Reply:
Yeah, is that what we want?

Reply:
We discussed this, and think it's less confusing to always show the static page for invalid sessions if the user has configured that.

Reply:
Discussed it and we concluded that it should never happen that a IP is invalid and if it is, we threat the requests same as non local ones